diff --git a/node_modules/astro/dist/core/routing/params.js b/node_modules/astro/dist/core/routing/params.js
index dfe6d5c..056bd83 100644
--- a/node_modules/astro/dist/core/routing/params.js
+++ b/node_modules/astro/dist/core/routing/params.js
@@ -1,7 +1,10 @@
-import { trimSlashes } from "../path.js";
+import { hasFileExtension, trimSlashes } from "../path.js";
 import { getRouteGenerator } from "./generator.js";
 import { validateGetStaticPathsParameter } from "./internal/validation.js";
 function stringifyParams(params, route, trailingSlash) {
+  if (route.type === "endpoint" && hasFileExtension(route.route)) {
+    trailingSlash = "never";
+  }
   const validatedParams = {};
   for (const [key, value] of Object.entries(params)) {
     validateGetStaticPathsParameter([key, value], route.component);
